src/ProxyKit/ProxyOptions.cs

```diff
@@ -7,5 +8,10 @@ public class ProxyOptions
         public TimeSpan? WebSocketKeepAliveInterval { get; set; }
 
         public int? WebSocketBufferSize { get; set; }
+        
+        /// <summary>
+        /// Determines the condition for copying request body to upstream. It is always `true` and copies regardless if not set.
+        /// </summary>
+        public Func<HttpRequest, bool> CopyRequestBodyIf { get; set; } = request => true;
```

Comment:
- Wondering if `Func<HttpRequest, bool>` should be `Predicate<T>` because it better communicates "this is a matching function" https://stackoverflow.com/questions/4317479/func-vs-action-vs-predicate#comment4689737_4317512
- Wonder if "CopyRequestBodyWhen" or just "CopyRequestBody" is better? (just thinking out loud).

Not to self:
 - xml comments on other properties
 - Default values and guard clauses on props (vs inferring such where consumed).